com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -222,6 +223,11 @@ public static boolean handleEvaluationResult(IDebugAdapterContext context, Threa
                 context.getProtocolServer().sendEvent(new Events.UserNotificationEvent(
                     Events.UserNotificationEvent.NotificationType.ERROR,
                     String.format("[Logpoint] Log message '%s' error: %s", breakpoint.getLogMessage(), ex.getMessage())));
+            } else if (value != null) {
+                if (value instanceof StringReference) {
+                    String message = ((StringReference) value).value();
+                    context.getProtocolServer().sendEvent(new Events.LogpointEvent(message));
```

Comment:
Instead of a custom LogpointEvent, I suggest to send a standard OutputEvent to the client. The client already supports displaying OutputEvent to debug console.

```java
Events.OutputEvent logpoint = new Events.OutputEvent(Events.OutputEvent.Category.console, message + System.lineSeparator());
context.getProtocolServer().sendEvent(logpoint);

```